### DISCUSSION TITLE
rename_relative, rename_basename and rename_root aren't working for me. none of them.

### DISCUSSION THREAD

**Discussion by Vinni-Cedraz** [nvim-tree/nvim-tree.lua#2368] [Q&A]
### Description

When I map "r" to either one of the three options I mentioned in the title what happens is that upon pressing `"r"` when the cursor is over a file the file opens in a new tab instead of a renaming option being prompted to me. I use the dressing.nvim plugin too by the way.

### Neovim version

```text
NVIM v0.9.0
Build type: Release
LuaJIT 2.1.0-beta3
```

### Operating system and version

Linux ubuntu 20.04 lts

### nvim-tree version

latest as updated by Lazy.nvim

### Clean room replication

```lua
view = {
		width = 30,
		side = "left",
		mappings = {
			list = {
				{ key = { "l", "<CR>", "o" }, cb = tree_cb("edit") },
				{ key = "h", cb = tree_cb("close_node") },
				{ key = "v", cb = tree_cb("vsplit") },
				{ key = "r", action = "rename_relative" },
			},
		},
	}
```
```


### Steps to reproduce

1 -> open nvim-tree
2 -> press the key "r" with the cursor over a filename

### Expected behavior

Being prompted to fill the new filename in order for it to be renamed.
Specifically, only the basename should show with the rename_root or rename_basename or rename_relative (was't clear for me which of them to use so I tried all of them) instead of the full path file name.

### Actual behavior

All that happens is the file under the cursor opens in a new tab, just like as if I was pressing enter or "o".

**Comment by alex-courtis** [comment]
`view.mappings.list` has been deprecated in favour of `on_attach`, please see [Migrating to on_attach](https://github.com/nvim-tree/nvim-tree.lua/wiki/Migrating-To-on_attach). You should have been notified on startup and pointed to the above page...

That action does... not exist, however there are many API functions to rename. See `:help nvim-tree-api.fs`

**Reply by Vinni-Cedraz** [reply]
According to issue #1155, where you participated in the solution, the action rename_relative should exist. Or any ormf the other names I mentioned. There was discussion between you and the guy who fixed the issue about the name of the action and the three names I mentioned were mentioned by you guys in there. It was not clear which one of them was chosen, that's why I tried them all.

**Reply by alex-courtis** [reply]
Please show the mappings you are using for your rename. Hopefully we can reproduce the issue.

**Comment by Vinni-Cedraz** [comment]
I already showed them! It's simply "r" when the cursor is over the filename
on nvim-tree.

**Reply by alex-courtis** [reply]
Please read [Migrating to on_attach](https://github.com/nvim-tree/nvim-tree.lua/wiki/Migrating-To-on_attach)

You need to use API.

**Reply by Vinni-Cedraz** [reply]
Ok, is it this one here? 
`api.fs.rename_basename`

**Reply by Vinni-Cedraz** [reply]
First thing in the guide you sent me tells me to run NvimTreeGenerateOnAttach command and this is what happened when I tried:

![image](https://github.com/nvim-tree/nvim-tree.lua/assets/92558763/90b9bc08-be94-460c-bfcc-b53737bdf707)

**Reply by gegoune** [reply]
This functionality has been removed recently. @alex-courtis shall we delete that wiki page?

**Reply by Vinni-Cedraz** [reply]
Then how else am I supposed to "migrate to on attach"?

**Reply by alex-courtis** [reply]
> This functionality has been removed recently. @alex-courtis shall we delete that wiki page?

It is time.

**Reply by alex-courtis** [reply]
> Then how else am I supposed to "migrate to on attach"?

You have had over 6 months to migrate and you have been repeatedly warned: every time nvim-tree starts.

As you only have 4 mappings, it is easiest just to create a new `on_attach` as per `:help nvim-tree-quickstart`

**Reply by Vinni-Cedraz** [reply]
No. I have been warned a few times 6 months ago, then I tried to do exactly what that wiki page suggested, which is running `:NvimTreeGenerateOnAttach` and that created a file that caused many conflicts because I was too much of a beginner to make it work so I ignored it and reverted onto standard configs and tried it again a few months after that only to receive the exact same message of "not an editor command" that I showed you. Then I did nothing since I *could do nothing* and was not having any issue until now and mostly because **I was not being warned everytime nvim-tree started** as you assumed. I was warned a few times 6 months ago that was it.

**Reply by alex-courtis** [reply]
```lua
local function my_on_attach(bufnr)
  local api = require("nvim-tree.api")

  local function opts(desc)
    return { desc = "nvim-tree: " .. desc, buffer = bufnr, noremap = true, silent = true, nowait = true }
  end

  -- default mappings
  api.config.mappings.default_on_attach(bufnr)

  -- custom mappings
  vim.keymap.set("n", "l", api.node.open.edit,             opts("Open"))
  vim.keymap.set("n", "h", api.node.navigate.parent_close, opts("Close Directory"))
  vim.keymap.set("n", "v", api.node.open.vertical,         opts("Open: Vertical Split"))
  vim.keymap.set("n", "r", api.fs.rename_basename,         opts("Rename"))
end
```

Perhaps you experienced https://github.com/neovim/neovim/issues/17832
